File: BackEnd/src/main/java/com/example/centerdeformation/model/Formation.java
package com.example.centerdeformation.model;

import lombok.Builder;
import lombok.Data;
import lombok.extern.jackson.Jacksonized;
import java.util.Date;

@Data
@Builder
@Jacksonized
public class Formation {

    private Long idFormation;

    private  String nomFormation;

    private float prix;

    private Date dateDebutFormation;

    private Date dateFinFormation;

    private String Horaire;

    private Modules listModules;

    private String photo;

    private EtatFormation etat;

}

File: FrontEnd/src/app/models/Formation.ts
import { Etudiant } from "./Etudiant/Etudiant";
import { Formateur } from "./Formateur/Formateur";
import { Modules } from "./Modules";

export class Formation {
    idFormation!: number;
    nomFormation:string='';
    prix!: number;
    dateDebutFormation!:Date;
    dateFinFormation!:Date;
    listeEtudiants!: Etudiant[];        
    horaire!:number;
    photo:string='';
    etat:string='';
    listeModules!: Modules[];
    nbEtudiant!:number;
  }
    